client/components/LearnerViewDirective.js

```diff
@@ -411,6 +470,8 @@ tie.directive('learnerView', [function() {
         };
 
         $scope.submitCode = function(code) {
+// LEFT OFF
+$scope.hideDotDiv = false;
           SolutionHandlerService.processSolutionAsync(
```

Comment:
I think it's due to this: http://stackoverflow.com/questions/38507733/angular-promise-hangs-ui

Basically, wrap SolutionHandlerService.processSolutionAsync in a $timeout with 0 delay so the execution of the $scope.hideDotDiv can happen first. I think things are hanging on the promise.